pkg/client/didexchange/client_test.go

```diff
@@ -1247,3 +1248,34 @@ func generateKeyPair() (string, []byte) {
 
 	return base58.Encode(pubKey[:]), privKey
 }
+
+func newPeerDID(t *testing.T) *did.Doc {
+	pubKey, _, err := ed25519.GenerateKey(rand.Reader)
+	require.NoError(t, err)
+
+	key := did.PublicKey{
+		ID:         uuid.New().String(),
+		Type:       "Ed25519VerificationKey2018",
+		Controller: "did:example:123",
+		Value:      pubKey,
+	}
+	doc, err := peer.NewDoc(
+		[]did.PublicKey{key},
```

Comment:
is `did.PublicKey` supposed to be the same as `did.VerificationMethod.PublicKey` as well as did.Service.RecipientKeys.

We need to start distinguishing encryption keys from verification keys. This is particularly important with the new Authcrypt/Anoncrypt packers which use valid KMS generated encryption keys. I'm interested to know how to integrate these keys along with DID verification keys to update didComm with the new Packers.

Reply:
@Baha-sk 

> is did.PublicKey supposed to be the same as did.VerificationMethod.PublicKey as well as did.Service.RecipientKeys.

No, but sadly there are a ton of assumptions already baked into the framework. Unpacking (pun) all of this will take effort.

> We need to start distinguishing encryption keys from verification keys. This is particularly important with the new Authcrypt/Anoncrypt packers which use valid KMS generated encryption keys. I'm interested to know how to integrate these keys along with DID verification keys to update didComm with the new Packers.

Yes. I'm happy to chat about it.

DIDComm - as it currently stands in Aries - is an odd case. It inlines the keys into the `did-communication` service entry. The location of these inlined keys may (probably will) change as the DIF/didcomm-messaging work evolves.